# Test SEED Keras

In [1]:
import os
os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
os.environ["CUDA_VISIBLE_DEVICES"] = ""

In [2]:
SEED = 123456

import random as rn
import numpy as np
import tensorflow as tf
 
os.environ['PYTHONHASHSEED']=str(SEED)
np.random.seed(SEED)
rn.seed(SEED)





session_conf = tf.compat.v1.ConfigProto(intra_op_parallelism_threads=1,
                              inter_op_parallelism_threads=1)



tf.random.set_seed(SEED)


In [3]:

sess = tf.compat.v1.Session(graph=tf.compat.v1.get_default_graph(),config=session_conf)
tf.compat.v1.keras.backend.set_session(sess)

In [5]:
from keras.models import Sequential
from keras.layers import Dense
import numpy

Using TensorFlow backend.


In [6]:
dataset = np.loadtxt("pima-indians-diabetes.csv", delimiter=",")

X = dataset[:,0:8]
Y = dataset[:,8]

In [7]:
model = Sequential()
model.add(Dense(12, input_dim=8, init='uniform', activation='relu'))
model.add(Dense(8, init='uniform', activation='relu'))
model.add(Dense(1, init='uniform', activation='sigmoid'))

/opt/conda/envs/tf2/lib/python3.7/site-packages/ipykernel_launcher.py:2: UserWarning: Update your `Dense` call to the Keras 2 API: `Dense(12, input_dim=8, activation="relu", kernel_initializer="uniform")`
  
/opt/conda/envs/tf2/lib/python3.7/site-packages/ipykernel_launcher.py:3: UserWarning: Update your `Dense` call to the Keras 2 API: `Dense(8, activation="relu", kernel_initializer="uniform")`
  This is separate from the ipykernel package so we can avoid doing imports until
/opt/conda/envs/tf2/lib/python3.7/site-packages/ipykernel_launcher.py:4: UserWarning: Update your `Dense` call to the Keras 2 API: `Dense(1, activation="sigmoid", kernel_initializer="uniform")`
  after removing the cwd from sys.path.


In [8]:
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])

In [9]:
model.fit(X, Y, epochs=2, batch_size=10, shuffle=False, verbose=0)

In [10]:
scores = model.evaluate(X, Y)
print("%s: %.2f%%" % (model.metrics_names[1], scores[1]*100))

768/768 [==============================] - 0s 166us/step
accuracy: 65.49%


Premiers tests
* Test 1 : accuracy: 77.86%
* Test 2 : accuracy: 78.91%
* Test 3 : accuracy: 78.12%
* Test 4 : accuracy: 79.04%

* Test 1 : accuracy: 79.56%
* Test 2 : accuracy: 79.95%
* Test 3 : accuracy: 80.47%


* Test 1 : accuracy: 79.82%
* Test 2 : accuracy: 81.12%
 

* Test 1 : accuracy: 79.17%


* Test 1 : accuracy: 71.35%
* Test 2 : accuracy: 77.21%



* Test 1 : accuracy: 78.91%
* Test 2 : accuracy: 80.73%

Test avec Seed env python, numpy random, random python et tensorflow
* Test 1 : accuracy: accuracy: 79.17%
* Test 2 : accuracy: accuracy: 79.17%
* Test 3 : accuracy: accuracy: 79.43%
* Test 4 : accuracy: accuracy: 79.30%
* Test 5 : accuracy: 80.21%

* Test 1 : accuracy: 80.73%
 

Test sans seed
* Test 1 : accuracy: 79.43%

* Test 2 : accuracy: 79.17%


* Test 3 : accuracy: 80.60%
* Test 4 : accuracy: 80.08%

Test 02/12/2019

avec shuffle= False

* Test 1 : accuracy: 78.78%
* Test 2 : accuracy: 80.73%
* Test 3 : accuracy: 78.12%
* Test 4 : accuracy: 

* Test 1 : accuracy: 79.17%
* Test 2 : accuracy: 79.30%
* Test 3 : accuracy: 79.17%
* Test 4 : accuracy: 78.26%
* Test 5 : accuracy: 72.27%

2 epochs avec shuffle= False 

* Test 1 : accuracy: 65.10%
* Test 2 : accuracy: 65.10%
* Test 3 : accuracy: 65.36%
* Test 4 : accuracy: 65.10%
* Test 5 : accuracy: 65.23%
* Test 6 : accuracy: 65.23%
* Test 7 : accuracy: 65.62%
* Test 8 : accuracy: 65.10%
* Test 9 : accuracy: 65.10%
* Test 1 : accuracy: 65.49%
